src/modules/navigator/rtl.cpp

```diff
@@ -56,6 +57,147 @@ RTL::on_inactive()
 {
 	// Reset RTL state.
 	_rtl_state = RTL_STATE_NONE;
+
+	find_RTL_destination();
+
+}
+
+void
+RTL::find_RTL_destination()
+{
+	// get home position:
+	home_position_s &home_landing_position = *_navigator->get_home_position();
+	// get global position
+	const vehicle_global_position_s &global_position = *_navigator->get_global_position();
+
+	switch (rtl_type()) {
+	case RTL_HOME: { // always take home position as landing destination
+			_destination.set(home_landing_position);
+			_destination_type = RTL_DESTINATION_HOME;
+			break;
+		}
+
+
+	case RTL_LAND: // take mission landing as landing destination (if available), otherwise home
+	case RTL_MISSION: { // inverse mission
+			if (_navigator->get_mission_start_land_available()) {
+				_destination.lat = _navigator->get_mission_landing_lat();
+				_destination.lon = _navigator->get_mission_landing_lon();
+				_destination.alt = _navigator->get_mission_landing_alt();
+				_destination_type = RTL_DESTINATION_MISSION_LANDING;
+
+			} else {
+				_destination.set(home_landing_position);
+				_destination_type = RTL_DESTINATION_HOME;
+			}
+
+			break;
+		}
+
+
+	case RTL_CLOSEST: { // choose closest possible landing point (consider home, mission landing and safe points)
+			_destination.set(
```

Comment:
Consider moving this block into a separate method to reduce the level of indentation.